lib/mongo/error/invalid_write_concern.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2014-2015 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Error
+
+    # Raised when an invalid write concern is provided.
+    #
+    # @since 2.2.0
+    class InvalidWriteConcern < Error
+
+      # Instantiate the new exception.
+      #
+      # @example Instantiate the exception.
+      #   Mongo::Error::InvalidWriteConcern.new
+      #
+      # @since 2.2.0
+      def initialize
+        super('Invalid write concern options.')
```

Comment:
It might be nice to provide a reason. In PyMongo the error message tells you why the write concern is invalid.